core/src/Cuda/Kokkos_Cuda_Graph_Impl.hpp

```diff
@@ -83,24 +85,31 @@ struct GraphImpl<Kokkos::Cuda> {
     //              destructor of the GraphImpl object (so that we don't have to
     //              just always do it)
     m_execution_space.fence("Kokkos::GraphImpl::~GraphImpl: Graph Destruction");
-    KOKKOS_EXPECTS(bool(m_graph))
+
```

Comment:
Why is that not something we should expect any more?
Why would it not be a requirement that the lifetime of the user-managed Cuda Graph must exceed of the Kokkos Graph that references it?

Reply:
I think what I had in mind is that if you build the graph from a `nullptr`, this line would fail.

But you're right. What I should test is that the `cudaGraph_t` passed to:
```c++
explicit GraphImpl(Kokkos::Cuda arg_instance, cudaGraph_t graph)
```
is valid (*i.e.* it's not *nullptr*). And I can re-establish `KOKKOS_EXPECTS(bool(m_graph))` in the constructor then.